carla-simulator/carla
# Windows setup and rebuild automatization

#### Description

"Setup.bat" and "Rebuild.cmd" automation for Windows platform.
`Setup.bat`: It downloads and installs Protobuf and Boost.
`Rebuild.bat`: Builds the project.

#### How to test?
1. Open the `Visual Studio x64 Native Tools Command Prompt` and navigate to the carla folder.
1. Run

       Setup.cmd -j 8 --boost-toolset msvc-14.1

   (Optional)

       Setup.cmd --help

   (You can see the available commands, for instance which `--boost-toolset` you have to use depending on the Visual Studio version that you have installed)

1. When it's done, download the Assets with the link provided when `Setup.cmd` finishes.
1. Finally run 

       Rebuild.bat

    that will compile carlaserver and start the Unreal project.

#### Where has this been tested?

  * **Platform(s):** Win10 x64
  * **Python version(s):** --
  * **Unreal Engine version(s):** 4.18.3

#### Possible Drawbacks

**[Linx]** CMake modified, it could be nice if someone test this branch in Linux.
**[Win]** Can make that already created projects fail (due to a new Boost file distribution)


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/carla-simulator/carla/429)
